lib/sycamore/sycamore/llms/prompts/default_prompts.py

```diff
@@ -1,64 +1,34 @@
-from abc import ABC, abstractmethod
 import logging
+from abc import ABC
 from typing import Optional, Type
 
-from guidance.models import Chat, Instruct, Model as GuidanceModel
-from guidance import gen, user, system, assistant, instruction
-
 logger = logging.getLogger(__name__)
 
 
-class GuidancePrompt(ABC):
-    @abstractmethod
-    def execute(self, model: GuidanceModel, **kwargs) -> str:
-        pass
+class SimplePrompt(ABC):
+    system: str = None
+    user: str = None
+    var_name: str = "answer"
+
+    def as_messages(self) -> list[dict]:
```

Comment:
Do we need to format here? I guess these messages would still have placeholders (like {filter_message}), right?

Reply:
This is only being used by the new prompts (the ones that format in the constructor). For guidance (which formats in execute()), this code isn't used. Maybe I'll leave a note

Reply:
But the 2 patterns are:

```
prompt = something # basically guidance prompts, has a .execute() implemented
prompt.execute(params=values_for_placeholders)
```

The other is:

```
prompt = Something(params=values_for_placeholders)
executor.execute(prompt)
executor.execute_w_messages_api(prompt.as_messages)
```